hs/src/LedgerState.hs

```diff
@@ -244,44 +237,39 @@ validDelegation cert l =
   <> validStakePoolRegister cert l
   <> validStakePoolRetire cert l
 
--- |Apply a raw transaction body as a state transition function on the ledger state.
-applyTx :: LedgerState -> Tx -> LedgerState
-applyTx ls tx =
-    LedgerState (txins tx </| getUtxo ls `union` txouts tx)
-                (getDelegationState ls)
-                (getEpoch ls)
-
 -- |In the case where a transaction is valid for a given ledger state,
 -- apply the transaction as a state transition function on the ledger state.
 -- Otherwise, return a list of validation errors.
 asStateTransition
-  :: Slot -> LedgerState -> LedgerEntry -> Either [ValidationError] LedgerState
-asStateTransition _ ls (TransactionData tx) =
+  :: Slot -> LedgerState -> TxWits -> Either [ValidationError] LedgerState
+asStateTransition slot ls tx =
   case valid tx ls of
```

Comment:
maybe we could also rename `valid` to a more specific name here? It validates the value transaction part